### ISSUE TITLE
openapi-to-typespec should not depend on a specific version of typespec

### ISSUE THREAD

**Issue by markcowl** [Azure/autorest#5047]
Because openapi-to-typespec is used by tsp-client, and tsp-client is a dependecy in the specs repo,  pinning the version in depenencies acauses multiple versions of typespec to be installed in the azure-rest-api-specs repo.

Ideally openapi-to-typespec would have a more open set of dependencies, similar to the typespec emitters.

This is somewhat urgent, because it has caused problems in two different TypeSpec releases.

**Comment by pshao25** [comment]
That's not doable. openapi-to-typespec generates code for that specific version of compiler. One copy of TypeSpec code cannot pass compilation to several versions of compilers. 

I suggest we do something at tsp-client side like "only when openapi-to-typespec is called then we install dependencies".

**Comment by mikeharder** [comment]
If the versions of `openapi-to-typespec` and `typespec` must match exactly, and repo `azure-rest-api-specs` depends on both, then I only see one solution.

Repo `azure-rest-api-specs` cannot upgrade either package, until matching versions of both are available.  Since we like to upgrade the specs repo as quickly as possible after typespec releases, this means a matching version of `openapi-to-typespec` should be ready at the same time (or very soon thereafter) as the `typespec` release.

@pshao25: How can we make this happen?

**Comment by pshao25** [comment]
Yes, if you could send me the TypeSpec change PR in `azure-rest-api-specs` several days before TypeSpec release, then I could get openapi-to-typespec prepared.
